### PR TITLE
[WIP]-Aadesha/port_validated_integ

### PR DESCRIPTION
**Pull request by Aadesha**
Closes https://github.com/JuliaReach/Reachability.jl/issues/611.

### CODE REVIEW

**Comment by mforets** [comment]
```julia
Reachability.solve_continuous: Error During Test at /home/travis/build/JuliaReach/Reachability.jl/test/Reachability/alltests.jl:2
  Got exception outside of a @test
  LoadError: UndefVarError: get_numvars not defined
```
-> the appropriate `using` is missing, in this case `using TaylorSeries: get_numvars`.

**Comment by mforets** [comment]
Please add `TaylorIntegration 0.4.1` in the [REQUIRE file](https://github.com/JuliaReach/Reachability.jl/blob/master/REQUIRE), in order to fix the warning:

```julia
┌ Warning: Package Reachability does not have TaylorIntegration in its dependencies:
│ - If you have Reachability checked out for development and have
│   added TaylorIntegration as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with Reachability
└ Loading TaylorIntegration into Reachability from project dependency, future warnings for Reachability are suppressed.
```

File: src/ReachSets/ContinuousPost/TMJets/reach.jl [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.=#
+
+using TaylorSeries:get_numvars
+using TaylorIntegration
```
Comment (mforets) [inline src/ReachSets/ContinuousPost/TMJets/reach.jl]: use `using TaylorIntegration: jetcoeffs!` here, and remove the module's name below (lines 83 and 86)

File: src/ReachSets/ContinuousPost/TMJets/reach.jl [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+using TaylorModels: validated_step!,TaylorModelN
+using IntervalArithmetic
```
Comment (mforets) [inline src/ReachSets/ContinuousPost/TMJets/reach.jl]:
```suggestion
import IntervalArithmetic
```

File: src/ReachSets/ContinuousPost/TMJets/reach.jl [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            aux = x[i](δt)
+            x[i]  = Taylor1( aux, orderT )
+            dx[i] = Taylor1( zero(aux), orderT )
+        end
+        # @show(IntervalBox(rem))
+        # @inbounds xTMNv[:, nsteps] .= xTMN[:]
+
+        # println(nsteps, "\t", t0, "\t", remainder.(xTMN[:]), "\t", diam(Δ))
+        if nsteps > maxsteps
+            @info("""
```
Comment (schillic) [inline src/ReachSets/ContinuousPost/TMJets/reach.jl]: It is better to do `info("...")` instead to use the logger.

File: src/ReachSets/ContinuousPost/TMJets/reach.jl [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    t   = t0 + Taylor1(orderT)
+    tI  = t0 + Taylor1(orderT+1)
+    δq_norm = IntervalBox(IntervalArithmetic.Interval{T}(-1,1),Val(N))
+    q0box = q0 + δq_norm
+
+    # Allocation of vectors
+    # Output
+    tv    = Array{T}(undef, maxsteps+1)
+    xv    = Array{IntervalBox{N,T}}(undef, maxsteps+1)
+    # xTMNv = Array{TaylorModelN{N,T,T}}(undef, dof, maxsteps+1)
```
Comment (schillic) [inline src/ReachSets/ContinuousPost/TMJets/reach.jl]: Could you remove the commented code in this file?

**Comment by mforets** [comment]
Thanks for the contribution!

This code is a replacement of `TaylorModels.validated_integ` so we need to add the copyright string, i think it's ok to add it in the same `reach.jl` file.

The test uses the tests for TMJets so it looks good. Also note that the build error reported in current master is not showing up (i cannot reproduce it either in my machine). Note, however, that a new warning is shown, about `update!` function. We can investigate it in a new issue.

In a subsequent PR we'll add early termination with invariant intersection.

File: src/ReachSets/ContinuousPost/TMJets/reach.jl (hunk 4 above)
Comment (mforets) [reply src/ReachSets/ContinuousPost/TMJets/reach.jl]: Yes. I'd like to make that and in general use our guidelines https://github.com/JuliaReach/JuliaReachDevDocs/

**Pull request merged**